fringe/src/fringe/utils/UIntLike.scala

```diff
@@ -0,0 +1,31 @@
+package fringe.utils
+
+import chisel3.{Data, UInt, Vec}
+
+trait UIntLike {
+  def asUInt: UInt
+  def raw = asUInt
+  def r = asUInt
+  implicit def conv: UInt = asUInt
+  def :=(other: UIntLike): Unit
+  def :=(other: UInt): Unit = {
+    val that = other
+    this := new UIntLike {
+      override def asUInt: UInt = that
+
+      // Don't need this I think
+      override def :=(other: UIntLike): Unit = Unit
+    }
+  }
+
+}
+
+class VecUInt[T <: Data](vec: Vec[T]) extends UIntLike {
+  override def asUInt: UInt = vec.asUInt()
+
+  override def :=(other: UIntLike): Unit = {
+    println(s"Assigning to VecUInt: ${vec}")
+    val otherUInt: UInt = other.asUInt
+    vec := otherUInt.asTypeOf(vec)
+  }
+}
```

Comment:
If I understand correctly, this means we can change my recent update for vecs in structs:

``` 
    case FieldApply(struct, field) =>
      emit(createWire(quote(lhs),remap(lhs.tp)))
      val (start, end) = getField(struct.tp, field)
      lhs.tp match {
        case v: Vec[_] => emit(src"$lhs.toSeq.zipWithIndex.foreach{case (fa, i) => fa := $struct($start, $end + (${v.size - 1} - i) * ${bitWidth(v.typeArgs.head)})}")
        case _ => emit(src"$lhs.r := $struct($start, $end)")
      }
```
to just `$lhs := $struct($start, $end)` for the case of a Vec field?